src/main/scala/com/databricks/labs/overwatch/pipeline/GoldTransforms.scala

```diff
@@ -490,131 +492,137 @@ trait GoldTransforms extends SparkSessionWrapper {
+        .filter('rnk === 1 && 'rn === 1).drop("rnk", "rn")
+
+      val joinedDF_after_before = joinedDF.filter(state_after_before)
+        .withColumn("sum_total_DBU_cost", sum(col("total_DBU_cost")).over(aggregationWindow))
+        .withColumn("sum_uptime_in_state_H", sum(col("uptime_in_state_H")).over(aggregationWindow))
+        .withColumn("rn", row_number.over(orderingWindow))
+        .withColumn("rnk", rank.over(orderingWindow))
+        .filter('rnk === 1 && 'rn === 1).drop("rnk", "rn")
+
+      val joinedDF_after = joinedDF.filter(state_after)
+        .withColumn("sum_total_DBU_cost", sum(col("total_DBU_cost")).over(aggregationWindow))
+        .withColumn("sum_uptime_in_state_H", sum(col("uptime_in_state_H")).over(aggregationWindow))
+        .withColumn("rn", row_number.over(orderingWindow))
+        .withColumn("rnk", rank.over(orderingWindow))
+        .filter('rnk === 1 && 'rn === 1).drop("rnk", "rn")
+
+      val joinedDF_before_after = joinedDF.filter(state_before_after)
+        .withColumn("sum_total_DBU_cost", sum(col("total_DBU_cost")).over(aggregationWindow))
+        .withColumn("sum_uptime_in_state_H", sum(col("uptime_in_state_H")).over(aggregationWindow))
+        .withColumn("rn", row_number.over(orderingWindow))
+        .withColumn("rnk", rank.over(orderingWindow))
+        .filter('rnk === 1 && 'rn === 1).drop("rnk", "rn")
+
+      val unionDF = joinedDF_before_before.union(joinedDF_after_before).union(joinedDF_after).union(joinedDF_before_after)
+        .withColumn("total_DBU_cost", sum(col("sum_total_DBU_cost")).over(aggregationWindow))
+        .withColumn("uptime_in_state_H", sum(col("sum_uptime_in_state_H")).over(aggregationWindow))
+        .withColumn("rn", row_number.over(orderingWindow))
+        .withColumn("rnk", rank.over(orderingWindow))
+        .drop("sum_total_DBU_cost", "sum_uptime_in_state_H")
+
+
+      val unionDF_final = unionDF.filter('rnk === 1 && 'rn === 1).drop("rnk", "rn")
+      val df = joinedDF.join(unionDF_final, Seq("unixTimeMSStart", "unixTimeMSEnd", "commandId"), "leftAnti")
+      val notebookClsfDF = unionDF_final.union(df.select(unionDF_final.schema.fieldNames.map(col): _*))
+      val notebookCodeAndMetaDF = notebookClsfDF.withColumnRenamed("unixTimeMS_state_start", "unixTimeMS")
+        .toTSDF("unixTimeMS", "organization_id", "date", "notebookId")
+        .lookupWhen(
+          notebookLookupTSDF,
+          maxLookAhead = 100
+        )
+        .df
+        .withColumn("dbu_cost_ps", col("total_DBU_cost") / col("uptime_in_state_H") / lit(3600))
+        .withColumn("estimated_dbu_cost", col("executionTime") * col("dbu_cost_ps"))
+        .drop("cluster_id")
+      notebookCodeAndMetaDF.select(colNames: _*)
+    }else{
+      println("Verbose Audit Log not enabled in workspace")
+      throw new VerboseAuditLoggingDisabled(s"Verbose Audit Logging is not enabled in the workspace. To get the data in notebookCommands_gold it is necessary you " +
```

Comment:
Can we use ModuleDisabled instead of defining a new Exception wrapper, in order to be consistent?

Reply:
@gueniai  Done